File: dynamic_listings/javascript/default/computer_science/enums2.js
// As of now, Javascript doesn't have native enums, so we use Object freezing to simulate it
const FUEL_TYPES = Object.freeze({
    GASOLINE: 1,
    DIESEL: 2,
    LPG: 3,
});

if (car.fuel_type === FUEL_TYPES.GASOLINE) {
    gasolinepump.use();
} else if (car.fuel_type === FUEL_TYPES.DIESEL) {
    dieselpump.use();
} else if (car.fuel_type === FUEL_TYPES.LPG) {
    lpgpump.use();
}

File: dynamic_listings/rust/default/computer_science/enums2.rs
#[derive(PartialEq, Eq)]
enum FuelType {
    Gasoline,
    Diesel,
    Lpg,
}
# struct Pump;
# impl Pump {
#    fn r#use(&self) {}
# }
# struct Car {
#     fuel_type: FuelType,
# }
# let car = Car { fuel_type: FuelType::Gasoline };
# let gasolinepump = Pump;
# let dieselpump = Pump;
# let lpgpump = Pump;

if car.fuel_type == FuelType::Gasoline {
    gasolinepump.r#use();
} else if car.fuel_type == FuelType::Diesel {
    dieselpump.r#use();
} else if car.fuel_type == FuelType::Lpg {
    lpgpump.r#use();
}
